api/src/main/java/io/keploy/service/GrpcService.java

```diff
@@ -155,6 +164,107 @@ public static void put(Service.TestCaseReq testCaseReq) {
         }
     }
 
+    /**
+     * Is valid test case to be recorded boolean.
+     * This method is used to check if the test case should be recorded or discarded.
+     *
+     * @param testCaseReq the test case req object which contains all the information about the test case
+     * @return the boolean value which indicates if the test case should be recorded or discarded
+     */
+    public static boolean isValidTestCaseToBeRecorded(Service.TestCaseReq testCaseReq) {
+        // acc    rej        record/discard
+        // true   true   =>  discard
+        // true   false  =>  record
+        // false  true   =>  discard
+        // false  false  =>  discard
+        return doesFollowAcceptanceRegex(testCaseReq) && !doesFollowRejectionRegex(testCaseReq);
+
+    }
+
+    /**
+     * Does follow acceptance regex boolean.
+     * This method is used to check if the test case url and header matches the acceptance regex.
+     *
+     * @param testCaseReq the test case req object which contains all the information about the test case
+     * @return the boolean value which indicates if the test case url and header matches the acceptance regex
+     */
```

Comment:
According to your `doesFollowAcceptanceRegex` -> if(acceptUrl || acceptHeader) return true. But it should be if(acceptURL && acceptHeader) return true.